src/tests/test_experiments.py

```diff
@@ -238,60 +265,63 @@ def test_validate_default(experiment_class):
         db_engine = create_engine(postgresql.url())
         populate_source_data(db_engine)
         with TemporaryDirectory() as temp_dir:
-            experiment = experiment_class(
-                config=sample_config(),
-                db_engine=db_engine,
-                project_path=os.path.join(temp_dir, "inspections"),
-                cleanup=True,
-            )
-            experiment.validate = mock.MagicMock()
-            experiment.run()
-            experiment.validate.assert_called_once()
+            with mock.patch("triage.component.catwalk.utils.open", side_effect=open_side_effect) as mock_file:
```

Comment:
I think you should be able to combine the nested `with` statements into one, along the lines of:
```python
with A() as a, B() as b:
    ...
```